shared/src/main/kotlin/watchfaces/dslclock/DSLClockComplication.kt

```diff
@@ -0,0 +1,355 @@
+fun DSLClockComplication() {
+
+    val style = remember {
+        TextStyle(
+            color = Color.White,
+            fontFamily = FontFamily.Monospace,
+            fontSize = 15.sp,
+        )
+    }
+
+    Background()
+    TimeAndOpenParentheses(style)
+    Date(style)
+    Battery(style)
+    HeartRate(style)
+    Steps(style)
+    Weather(style)
+    CloseParentheses(style)
+
+}
+
+
+@Composable
+private fun Background() {
+    val isAmbient by LocalIsAmbient.current
+    OClockCanvas {
+        if (isAmbient.not()) drawCircle(kotlinDarkBg)
+    }
+}
+
+@Composable
+private fun TimeAndOpenParentheses(style: TextStyle) {
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "complications {",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 60f,
+                    y = (center.y * 0.4f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+}
+
+
+@Composable
+private fun Date(style: TextStyle) {
+
+    val formatter = DateTimeFormatter.ofPattern("MMM dd, yyyy")
+    val date = LocalDate.now().format(formatter)
+
+    @OptIn(InternalComposeOClockApi::class)
+    val measurer = LocalTextMeasurerWithoutCache.current
+
+    val textLayoutResult by remember {
+        derivedStateOf {
+            measurer.measure(
+                "date = $date",
+                style
+            )
+        }
+    }
+
+    OClockCanvas {
+        rotate(degrees = 0f) {
+            drawText(
+                brush = style.brush ?: SolidColor(style.color),
+                textLayoutResult = textLayoutResult,
+                topLeft = Offset(
+                    x = 120f,
+                    y = (center.y * 0.6f) - (textLayoutResult.size.height / 2f)
+                )
+            )
+        }
+    }
+
+}
+
+
+@Composable
+private fun Battery(style: TextStyle) {
+
+    val context: Context = LocalContext.current
+    var batteryLevel by remember { mutableIntStateOf(0) }
+
+    val iFilter = IntentFilter(Intent.ACTION_BATTERY_CHANGED)
+    val batteryStatus = context.registerReceiver(null, iFilter)
+
+    batteryLevel = batteryStatus?.getIntExtra(BatteryManager.EXTRA_LEVEL, -1) ?: -1
+
+    context.registerReceiver(
+        object : BroadcastReceiver() {
+            override fun onReceive(ctxt: Context, intent: Intent) {
+                try {
+                    batteryLevel = intent.getIntExtra(BatteryManager.EXTRA_LEVEL, -1)
+                } catch (_: IllegalStateException) {
+                }
+            }
+        },
+        IntentFilter(Intent.ACTION_BATTERY_CHANGED))
```

Comment:
You wouldn't want to run this code on every recomposition.
Most of that needs to be put inside a `LaunchedEffect`, a `DisposableEffect`, or a `produceState`, as usual in Compose.